core/processor/abci.go

```diff
@@ -2635,3 +2647,32 @@ func (app *App) JoinTeam(ctx context.Context, tx abci.Tx) error {
 
 	return nil
 }
+
+func (app *App) CheckUpdatePartyProfile(ctx context.Context, tx abci.Tx) error {
+	params := &commandspb.UpdatePartyProfile{}
+	if err := tx.Unmarshal(params); err != nil {
+		return fmt.Errorf("could not deserialize UpdatePartyProfile command: %w", err)
+	}
+
+	partyID := types.PartyID(tx.Party())
+	if err := app.partiesEngine.CheckProfile(ctx, partyID, params); err != nil {
+		return err
+	}
```

Comment:
We can'd do that here, this being called by checkTx can be called concurrently, because of that, if I'm not mistaken, we could read from the map used in the parties engine whicle calling checkTx and processing a PartiesUpdates transaction. While that's very unlikely it happen in real life, I'd rather not risk it.